js/src/annotations/osd-region-draw-tool.js

```diff
@@ -76,37 +76,38 @@
       }
     },
 
-    render: function() {
+    render: function () {
+
+      if(this.parent.mode !== $.AnnotationsLayer.DISPLAY_ANNOTATIONS){
+        return ;
+      }
       this.svgOverlay.restoreEditedShapes();
       this.svgOverlay.paperScope.activate();
       this.svgOverlay.paperScope.project.clear();
       var _this = this;
       _this.annotationsToShapesMap = {};
-      var deferreds = jQuery.map(this.list, function(annotation) {
-        var deferred = jQuery.Deferred(),
-        shapeArray;
+
+      for (var i = 0; i < this.list.length; i++) {
```

Comment:
Why not this.list.forEach() ? It's much more readable, and matches the style of the rest of the codebase.